cni/pkg/iptables/iptables.go

```diff
@@ -259,14 +257,38 @@ func (cfg *IptablesConfigurator) appendInpodRules(hostProbeSNAT *netip.Addr) *bu
 	//
 	// DESC: Anything coming BACK from the pod healthcheck port with a dest of our SNAT-ed hostside IP
 	// we also short-circuit.
-	iptablesBuilder.AppendRule(
+	iptablesBuilder.AppendRuleV4(
 		iptableslog.UndefinedCommand, ChainInpodOutput, iptablesconstants.NAT,
 		"-d", hostProbeSNAT.String(),
 		"-p", "tcp",
 		"-m", "tcp",
 		"-j", "ACCEPT",
 	)
 
+	if cfg.cfg.EnableIPv6 {
+		// CLI: -t mangle -A ISTIO_PRERT -s fd16:9254:7127:1337:ffff:ffff:ffff:ffff -p tcp -m tcp --dport <PROBEPORT> -j ACCEPT
+		//
+		// DESC: If this is one of our node-probe ports and is from our SNAT-ed/"special" hostside IP, short-circuit out here
+		iptablesBuilder.AppendRuleV6(iptableslog.UndefinedCommand, ChainInpodPrerouting, iptablesconstants.MANGLE,
```

Comment:
any desire to use AppendVersionedRule?

Reply:
Ah that's right - I explicitly didn't for the hostside variants, but we should here - ty!

Reply:
Actually, it doesn't save any code here, since this is additive (always do v4, maybe do v6) - AppendVersionedRule is a bit limiting - it's not always JUST the IP that's different between V4 and V6 rules, and I kinda hate "dualstack" as a concept anyway - you either support v6, or you don't, it isn't inherently related to v4 support.

Reply:
I don't care too much but isn't it just the IP different here? and AppendVersionedRule already handles not adding the rule if v6 isn't enabled (actually, so does AppendRuleV6, so strictly speaking you don't _need_ the If on L268?)?

Reply:
> AppendVersionedRule already handles not adding the rule if v6 isn't enabled

Oh it does, a few levels down. Missed that. That's fair, it would save the `if` here.